Index: dpkg/ostable
===================================================================
--- dpkg.orig/ostable	2009-01-26 05:59:29.000000000 +0600
+++ dpkg/ostable	2009-01-26 06:00:56.000000000 +0600
@@ -25,3 +25,5 @@
 bsd-netbsd	netbsd		netbsd[^-]*
 bsd-openbsd	openbsd		openbsd[^-]*
 sysv-solaris	solaris		solaris[^-]*
+uclibc-linux	linux-uclibc	linux[^-]*-uclibc
+uclibceabi-linux	linux-uclibceabi	linux[^-]*-uclibceabi
Index: dpkg/triplettable
===================================================================
--- dpkg.orig/triplettable	2009-01-26 05:59:29.000000000 +0600
+++ dpkg/triplettable	2009-01-26 06:01:02.000000000 +0600
@@ -14,3 +14,7 @@
 bsd-netbsd-<cpu>	netbsd-<cpu>
 bsd-darwin-<cpu>	darwin-<cpu>
 sysv-solaris-<cpu>	solaris-<cpu>
+uclibc-linux-<cpu>	uclibc-<cpu>
+uclibceabi-linux-arm	uclibc-armel
+uclibceabi-linux-armeb	uclibc-armeb
+gnueabi-linux-armeb	armeb
