--- a/scripts/dpkg-buildpackage.pl
+++ b/scripts/dpkg-buildpackage.pl
@@ -51,7 +51,7 @@
   -spgp          the sign-command is called like PGP.
   -us            unsigned source.
   -uc            unsigned changes.
-  -a<arch>       Debian architecture we build for (implies -d).
+  -a<arch>       Debian architecture we build for.
   -b             binary-only, do not build source.   } also passed to
   -B             binary-only, no arch-indep files.   } dpkg-genchanges
   -A             binary-only, only arch-indep files. }
@@ -138,7 +138,6 @@
 	$usepause = 1;
     } elsif (/^-a(.*)$/) {
 	$targetarch = $1;
-	$checkbuilddep = 0;
     } elsif (/^-s[iad]$/) {
 	$sourcestyle = $_;
     } elsif (/^-s[nsAkurKUR]$/) {
