--- a/debian/dpkg.install
+++ b/debian/dpkg.install
@@ -14,7 +14,6 @@
 usr/lib/dpkg/mksplit
 usr/sbin
 usr/share/dpkg
-usr/share/locale/*/LC_MESSAGES/dpkg.mo
 usr/share/man/{*/*,*}/dpkg-deb.1
 usr/share/man/{*/*,*}/dpkg-divert.8
 usr/share/man/{*/*,*}/dpkg-query.1
--- a/man/Makefile.am
+++ b/man/Makefile.am
@@ -19,8 +19,6 @@
 
 # FIXME: Use a stamp file until po4a supports them internally.
 man.stamp:
-	$(PO4A_V) cd $(srcdir) && \
-	po4a $(PO4A_BUILD_OPTS) po/po4a.cfg
 	$(AM_V_at) touch $@
 
 clean-local: clean-local-@USE_NLS@
